File: cases/taint/php/return-relay-one-hop-negative/return_relay_one_hop_negative.php
<?php
function dfb_source(): string { // DFB-SOURCE: return-one-hop-negative-input
    return "tainted";
}

function relay(string $value): string { // DFB-WITNESS: return-one-hop-negative-relay
    return $value;
}

function dfb_sink(string $value): void {} // DFB-SINK: return-one-hop-negative-sink

function run(): void {
    $result = relay(dfb_source());
    dfb_sink("clean");
}

File: cases/taint/cpp/return-relay-one-hop-negative/return_relay_one_hop_negative.cpp
int dfb_source() { // DFB-SOURCE: return-one-hop-negative-input
    return 1;
}

int relay(int value) { // DFB-WITNESS: return-one-hop-negative-relay
    return value;
}

void dfb_sink(int value) {} // DFB-SINK: return-one-hop-negative-sink

void run() {
    int result = relay(dfb_source());
    dfb_sink(0);
}
